firewall.go

```diff
@@ -256,3 +277,59 @@ func (c *Client) GetFirewallEventList(id string) ([]Event, error) {
 	}
 	return firewallEvents, err
 }
+
+//waitForFirewallActive allows to wait until the firewall's status is active
+func (c *Client) waitForFirewallActive(id string) error {
+	timer := time.After(c.cfg.requestCheckTimeoutSecs)
+	delayInterval := c.cfg.delayInterval
+	for {
+		select {
+		case <-timer:
+			errorMessage := fmt.Sprintf("Timeout reached when waiting for firewall %v to be active", id)
+			c.cfg.logger.Error(errorMessage)
+			return errors.New(errorMessage)
+		default:
+			time.Sleep(delayInterval) //delay the request, so we don't do too many requests to the server
+			fw, err := c.GetFirewall(id)
+			if err != nil {
+				return err
+			}
+			if fw.Properties.Status == activeStatus {
+				return nil
+			}
+		}
+	}
+}
+
+//waitForFirewallDeleted allows to wait until the firewall is deleted
+func (c *Client) waitForFirewallDeleted(id string) error {
+	if !isValidUUID(id) {
+		return errors.New("'id' is invalid")
+	}
+	timer := time.After(c.cfg.requestCheckTimeoutSecs)
+	delayInterval := c.cfg.delayInterval
+	for {
+		select {
+		case <-timer:
+			errorMessage := fmt.Sprintf("Timeout reached when waiting for firewall %v to be deleted", id)
+			c.cfg.logger.Error(errorMessage)
+			return errors.New(errorMessage)
+		default:
+			time.Sleep(delayInterval) //delay the request, so we don't do too many requests to the server
+			r := Request{
+				uri:          path.Join(apiFirewallBase, id),
+				method:       http.MethodGet,
+				skipPrint404: true,
```

Comment:
what's the benefit of skipPrint404?

Reply:
Sometime we use 404 to check existence of an object so that we don't want this to be printed as an error